File: RedditPictureReader/app/src/main/java/se/slotte/ludwig/redditpicturereader/picture_feed/PicturePresenter.java
package se.slotte.ludwig.redditpicturereader.picture_feed;

import java.util.List;

import se.slotte.ludwig.redditpicturereader.picture_feed.data.FetchAllDataPresenter;
import se.slotte.ludwig.redditpicturereader.picture_feed.data.model.Children;

/**
 * Created by ludwigslotte on 16/10/16.
 */

class PicturePresenter implements FetchAllDataPresenter.FetchDataCallback {
    private PictureView view;
    private FetchAllDataPresenter fetchAllDataPresenter = new FetchAllDataPresenter();

    PicturePresenter(PictureView view) {
        this.view = view;
    }

    void fetchPhotos() {
        view.loadingAnimation();
        fetchAllDataPresenter.fetchPhotos(this);
    }

    @Override
    public void onComplete(List<Children> success) {
        view.onSuccess(success);
    }

    @Override
    public void onNetworkError() {
        view.onNetworkError();
    }
}

File: RedditPictureReader/app/src/main/java/se/slotte/ludwig/redditpicturereader/picture_feed/PicturePresenter.kt
package se.slotte.ludwig.redditpicturereader.picture_feed

import se.slotte.ludwig.redditpicturereader.picture_feed.data.FetchAllData
import se.slotte.ludwig.redditpicturereader.picture_feed.data.FetchDataCallback
import se.slotte.ludwig.redditpicturereader.picture_feed.data.model.Children

class PicturePresenter(val view: PictureView) : FetchDataCallback {
    private val fetchAllDataPresenter = FetchAllData()

    fun init(){
        view.setOnErrorClickListener()
        fetchPhotos()
    }

    fun fetchPhotos() {
        view.startLoading()
        fetchAllDataPresenter.fetchPhotos(this)
    }

    override fun onComplete(success: List<Children>?) {
        view.onSuccess(success)
    }

    override fun onNetworkError() {
        view.onNetworkError()
    }

    override fun onGeneralError() {
        view.onGeneralError()
    }
}
